task17/index.js

```diff
@@ -0,0 +1,127 @@
+
+const slideShowContainer = document.getElementById("js-slideShowContainer");
+const slideShowWrapper = document.getElementById("js-slideShowWrapper");
+const slideImgList = document.getElementById("js-slideImgList");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const addLoading = () => {
+  const loadingPlace = createElementWithClassName("div", "loader");
+  loadingPlace.id = "js-loadingPlace";
+  slideShowWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("js-loadingPlace").remove();
+
+const renderErrorMessage = (errorMessage) => {
+  slideShowWrapper.textContent = errorMessage;
+};
+
+const getFetchData = async () => {
+  const response = await fetch(URL);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage("問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImgData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(URL));
+    }, 3000);
+  });
+};
+
+const init = async () => {
+  addLoading();
+  let slideImgData;
+  try {
+    slideImgData = await fetchImgData();
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
+  } finally {
+    removeLoading();
+  }
+  currentImgIndex = slideImgData.findIndex((v) => v.display);
+  renderSlideItem(slideImgData);
+  renderSlideNumber(slideImgData);
+  addEventListenerForBtn();
+  toggleDisabledOfButton(slideImgData);
+};
+init();
+
+const renderSlideItem = (slideImgData) => {
+  const slideImgFragment = document.createDocumentFragment();
+  for (let i = 0; i < slideImgData.length; i++) {
+    const slideImgItem = createElementWithClassName("li", "slide-img-item");
+    const slideImage = createElementWithClassName("img", "slide-image");
+    slideImage.src = slideImgData[i].img;
+    slideImage.alt = slideImgData[i].alt;
+    slideImgData[i].display && slideImgItem.classList.add("is-active");
+    slideImgFragment.appendChild(slideImgItem).appendChild(slideImage);
+  }
+  slideImgList.appendChild(slideImgFragment);
+  createSlideBtn();
+};
+
+const createSlideBtn = () => {
+  const slideShowItem = document.getElementById("js-slideShowItem");
+  const prevButton = createElementWithClassName("button", "slide-btn");
+  const nextButton = createElementWithClassName("button", "slide-btn");
+  const prevBtnImg = document.createElement("img");
+  const nextBtnImg = document.createElement("img");
+  prevButton.id = "js-prevBtn";
+  nextButton.id = "js-nextBtn";
+  prevBtnImg.src = "./img/prev-arrow.png";
+  nextBtnImg.src = "./img/next-arrow.png";
+  prevButton.appendChild(prevBtnImg);
+  slideShowItem.insertBefore(prevButton, slideShowWrapper);
+  slideShowItem.appendChild(nextButton).appendChild(nextBtnImg);
+};
```

Comment:
IMO
It is also possible to use an array and create elements in a loop as follows.
```suggestion
const createSlideBtn = () => {
  const directions = ["prev", "next"];
  directions.forEach((direction) => {
    const button = createElementWithClassName("button", "slide-btn");
    const img = document.createElement("img");
    button.id = `js-${direction}Btn`;
    img.src = `./img/${direction}-arrow.png`;
    button.append(img);
    direction === "prev" && slideShowWrapper.before(button);
    direction === "next" && slideShowWrapper.after(button);
  })
};
```

Reply:
You can hang it like this! Thanks for letting me know!
I am very impressed with the review by Ms. Sae, and I am very impressed with how it turned out to be so different from my own code.
d24c5c2